### DISCUSSION TITLE
Localization and stringFormat

### DISCUSSION THREAD

**Discussion by acaliaro** [SirJohnK/LocalizationResourceManager.Maui#18] [General]
Hi
is it possible to user LocalizationManager with stringFormat directly in XAML?

https://github.com/dotnet/docs-maui/blob/main/docs/fundamentals/data-binding/string-formatting.md

**Comment by SirJohnK** [comment]
Sure! The ```TranslateExtension``` supports StringFormat.
```csharp
            <Label
                FontSize="18"
                HorizontalOptions="Center"
                SemanticProperties.Description="{localization:Translate WelcomeToMAUI}"
                SemanticProperties.HeadingLevel="Level2"
                Text="{localization:Translate WelcomeToMAUI, StringFormat='Yo, {0}'}" />
```
...or if you want to inject values into a translated text, this will come in the upcoming release with the PR #16!

**Comment by acaliaro** [comment]
Thanks @SirJohnK but I have not understand.
I would like something like

Text="{Binding MyProperty, StringFormat='localization:Translate MyPropertyTranslatedFormatString'}"

Where

MyPropertyTranslatedFormatString = 'Hi {0}'

**Reply by SirJohnK** [reply]
Yes, this will come in the next release (v1.2.0) with the ```TranslateBinding``` in #16! (Will probably release that tonight!)

**Reply by acaliaro** [reply]
I will not sleep, thanks
Alessandro

**Comment by SirJohnK** [comment]
This feature is now live on [NuGet](https://www.nuget.org/packages/LocalizationResourceManager.Maui/1.2.0) in the latest release [v1.2.0](https://github.com/SirJohnK/LocalizationResourceManager.Maui/releases/tag/1.2.0)!

See more information in the [Readme section](https://github.com/SirJohnK/LocalizationResourceManager.Maui/blob/main/README.md)!

With the `TranslateBindingExtension` you can now do this:
```csharp
Text="{localization:TranslateBinding MyProperty, TranslateFormat=MyPropertyTranslatedFormatString}"
```

**Comment by SirJohnK** [comment]
@acaliaro , have you seen the new release!? :smiley:

**Reply by acaliaro** [reply]
Not yet, I hope next week..
